src/projections/igh_o.cpp

```diff
@@ -0,0 +1,254 @@
+    const double y90 = Q->dy0 + sqrt(2.0); /* lt=90 corresponds to y=y0+sqrt(2) */
+
+    int z = 0;
+    if (xy.y > y90+EPSLN || xy.y < -y90+EPSLN) /* 0 */
+      z = 0;
+    else if (xy.y >=  d4044118)
+           if (xy.x <=  -d110) z =  1;
+      else if (xy.x >=    d50) z =  3;
+      else z = 2;
+    else if (xy.y >=  0)
+           if (xy.x <=  -d110) z =  4;
+      else if (xy.x >=    d50) z =  6;
+      else z = 5;
+    else if (xy.y >= -d4044118) {
+           if (xy.x <=   -d75) z =  7;
+      else if (xy.x >=    d90) z =  9;
+      else z = 8;
+    }
+    else {
+           if (xy.x <=   -d75) z =  10;
+      else if (xy.x >=    d90) z =  11;
+      else z = 12;
+    }
+
+    if (z) {
+        int ok = 0;
+
+        xy.x -= Q->pj[z-1]->x0;
+        xy.y -= Q->pj[z-1]->y0;
+        lp = Q->pj[z-1]->inv(xy, Q->pj[z-1]);
+        lp.lam += Q->pj[z-1]->lam0;
+
+        switch (z) {
+        /* Extend the plottable area slightly for each sub-projection */
+        case  1:  ok = (lp.lam  >= -d180-EPSLN && lp.lam <= -d110+EPSLN) ||
+                       ((lp.lam >= -d180-EPSLN && lp.lam <=  -d70+EPSLN) &&
+                       (lp.phi  >=   d60-EPSLN && lp.phi <=   d90+EPSLN)); break;
+        case  2:  ok = (lp.lam  >= -d110-EPSLN && lp.lam <=   d50+EPSLN) ||
+                       ((lp.lam >=   d50-EPSLN && lp.lam <=   d60+EPSLN) &&
+                       (lp.phi  >=   d60-EPSLN && lp.phi <=   d90+EPSLN)) ||
+                       ((lp.lam <= -d120-EPSLN && lp.lam >= -d110+EPSLN) &&
+                       (lp.phi  >=   d60-EPSLN && lp.phi <=   d90+EPSLN)) ; break;
+        case  3:  ok = (lp.lam  >=   d50-EPSLN && lp.lam <=  d180+EPSLN) ||
+                       ((lp.lam >=   d10-EPSLN && lp.lam <=   d60+EPSLN) &&
+                       (lp.phi  >=   d60-EPSLN && lp.phi <=   d90+EPSLN)); break;
+        case  4:  ok = (lp.lam  >= -d180-EPSLN && lp.lam <= -d110+EPSLN); break;
+        case  5:  ok = (lp.lam  >= -d110-EPSLN && lp.lam <=   d50+EPSLN); break;
+        case  6:  ok = (lp.lam  >=   d50-EPSLN && lp.lam <=  d180+EPSLN); break;
+        case  7:  ok = (lp.lam  >= -d180-EPSLN && lp.lam <=  -d75+EPSLN); break;
+        case  8:  ok = (lp.lam  >=  -d75-EPSLN && lp.lam <=   d90+EPSLN); break;
+        case  9:  ok = (lp.lam  >=   d90-EPSLN && lp.lam <=  d180+EPSLN); break;
+        case  10: ok = (lp.lam  >= -d180-EPSLN && lp.lam <=  -d75+EPSLN); break;
+        case  11: ok = (lp.lam  >=  -d75-EPSLN && lp.lam <=   d90+EPSLN); break;
+        case  12: ok = (lp.lam  >=   d90-EPSLN && lp.lam <=  d180+EPSLN); break;
+
+        }
+      z = (!ok? 0: z); /* projectable? */
+    }
+
+    if (!z) lp.lam = HUGE_VAL;
+    if (!z) lp.phi = HUGE_VAL;
+
+    return lp;
+}
+
+
+static PJ *destructor (PJ *P, int errlev) {
+    int i;
+    if (nullptr==P)
+        return nullptr;
+
+    if (nullptr==P->opaque)
+        return pj_default_destructor (P, errlev);
+
+    for (i = 0; i < 12; ++i) {
```

Comment:
```suggestion
    struct pj_opaque *Q = static_cast<struct pj_opaque*>(P->opaque);
    for (i = 0; i < 12; ++i) {
```
and replace all 3 below uses of ``static_cast<struct pj_opaque*>(P->opaque)`` with ``Q``

Reply:
Thanks. Done.